internal/controller/providers/pxc/applier.go

```diff
@@ -102,6 +109,55 @@ func configureStorage(
 	return common.ConfigureStorage(ctx, c, db, currentSize, setStorageSize)
 }
 
+// generatePass generates a random password.
+func generatePass() ([]byte, error) {
+	randLenDelta, err := rand.Int(rand.Reader, big.NewInt(int64(passwordMaxLen-passwordMinLen)))
+	if err != nil {
+		return nil, err
+	}
+
+	b := make([]byte, passwordMinLen+randLenDelta.Int64())
+	for i := range b {
+		randInt, err := rand.Int(rand.Reader, big.NewInt(int64(len(passSymbols))))
+		if err != nil {
+			return nil, err
+		}
+		b[i] = passSymbols[randInt.Int64()]
+	}
+
+	return b, nil
+}
+
+func (p *applier) ensureSecretHasProxyAdminPassword(secretsName string) error {
+	secret := &corev1.Secret{}
+	err := p.C.Get(p.ctx, types.NamespacedName{
+		Name:      p.DB.Spec.Engine.UserSecretsName,
+		Namespace: p.DB.GetNamespace(),
+	}, secret)
+	if client.IgnoreNotFound(err) != nil {
+		return err
+	}
+
+	if _, ok := secret.Data["proxyadmin"]; ok {
+		return nil
+	}
```

Comment:
Is it possible for the pxc-operator to set this password before everest-operator? In that case, the key would still exist with the buggy password, and we would allow it because we're just checking for the existence for the key

Reply:
I'm pretty sure it's guaranteed that this code runs before creating the PXC CR in the first reconciliation so I don't think that there's a chance for the PXC operator to mess with this secret.